js/json.test.ts

```diff
@@ -105,3 +107,56 @@ test("Valid JSON Test", async () => {
     expect(score).toEqual(expected);
   }
 });
+
+test("Semantic JSON Test", async () => {
+  const cases = [
+    { a: '{"x": 1, "y": 2}', b: '{"y": 2, "x": 1}', expected: 1 },
+    {
+      a: '{"zs": ["a", "b"], "x": 1, "y": 2}',
+      b: '{"y": 2, "zs": ["a", "b"], "x": 1}',
+      expected: 1,
+    },
+    {
+      a: '{"o1": {"x": 1, "y": 2}}',
+      b: '{"o1": {"y": 2, "x": 1}}',
+      expected: 1,
+    },
+    {
+      a: '{"xs": [{"o1": {"x": 1, "y": [2]}}]}',
+      b: '{"xs": [{"o1": {"y": [2], "x": 1}}]}',
+      expected: 1,
+    },
+    {
+      a: '{"o1": {"x": 2, "y": 2}}',
+      b: '{"o1": {"y": 2, "x": 1}}',
+      expected: 0.83333,
+    },
+    {
+      a: '{"o1": {"x": 2, "y": 2}}',
```

Comment:
This test case is identical to the one above.

Reply:
Thanks for flagging, just tweaked it